core/server/api/canary/users.js

```diff
@@ -230,5 +233,31 @@ module.exports = {
                 return models.ApiKey.refreshSecret(model.toJSON(), Object.assign({}, {id: model.id}));
             });
         }
+    },
+
+    resetAllPasswords: {
+        headers: {
+            cacheInvalidate: true
+        },
+        permissions: async (frame) => {
+            // TODO: Is there an easier way than querying the roles?
+            const user = await models.User.findOne({
+                id: frame.user.get('id'),
+                status: 'all'
+            }, {withRelated: ['roles']});
+            const isAdmin = user.hasRole('Owner') || user.hasRole('Administrator');
+            if (!isAdmin) {
+                return Promise.reject();
+            }
+            return Promise.resolve();
+        },
+        async query(frame) {
+            await userService.resetAllPasswords(frame.options);
+
+            // TODO: make sure it's ok to return this kind of JSON
```

Comment:
Would be good to have references for this from other endpoints that are "actions" and not resource fetching.

Reply:
The only other instances I could find were [these](https://github.com/TryGhost/Ghost/blob/f4017cc4392afee117a29ad11d47a76e3883799e/core/server/api/canary/members.js#L346
) [two](https://github.com/TryGhost/Ghost/blob/273e22032761d39aef10e0b9a90584f31b0271c9/core/server/api/canary/webhooks.js#L146) so I'm changing this to `null` .

Reply:
Yup. This is way better! :+1: